ribir/src/winit_shell_wnd.rs

```diff
@@ -177,15 +177,36 @@ pub(crate) fn new_id(id: winit::window::WindowId) -> WindowId {
 }
 
 impl WinitShellWnd {
+  #[cfg(target_family = "wasm")]
+  pub(crate) async fn new_with_canvas<T>(
```

Comment:
We should not define a `new_with_canvas` method. Instead, keep the same signature as `new`, but implement it differently.